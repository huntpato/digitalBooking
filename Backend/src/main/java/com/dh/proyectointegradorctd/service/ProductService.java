package com.dh.proyectointegradorctd.service;

import com.dh.proyectointegradorctd.model.Category;
import com.dh.proyectointegradorctd.model.City;
import com.dh.proyectointegradorctd.model.Product;
import com.dh.proyectointegradorctd.model.User;
import com.dh.proyectointegradorctd.repository.IProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService implements IEntityService<Product>{

    /* ======================= Atributos ====================== */

    private final IProductRepository productRepository;
    private final CategoryService categoryService;
    private final CityService cityService;
    private final UserService userService;

    /* ======================= MÃ©todos ======================== */

    @Override
    public Product save(Product product) {

        Category cat = categoryService.findCategoryByCategory(product.getProductCategory().getName());
        User us = userService.findByEmail(product.getUser().getEmail());
        City city = cityService.findCityByCountryAndName(product.getCity().getCountry(), product.getCity().getName());

        if(cat != null){
            product.setProductCategory(cat);
        }

        if(us != null){
            product.setUser(us);
        }

        if(city != null){
            product.setCity(city);
        }

        return productRepository.save(product);

    }

    @Override
    public List<Product> findAll() {
        return productRepository.findAll();
    }

    @Override
    public Product findById(Integer id) {
        return productRepository.findById(id).orElse(null);
    }

    @Override
    public Product update(Product product) {

        Product tmp = productRepository.findById(product.getId()).orElse(null);

        City city = cityService.findCityByCountryAndName(product.getCity().getCountry(), product.getCity().getName());
        Category cat = categoryService.findCategoryByCategory(product.getProductCategory().getName());
        User us = userService.findByEmail(product.getUser().getEmail());

        if(cat != null){
            product.setProductCategory(cat);
        }

        if(us != null){
            product.setUser(us);
        }

        if(city != null){
            product.setCity(city);
        }

        if(tmp != null){

            tmp.setProductName(product.getProductName());
            tmp.setProductDescription(product.getProductDescription());
            tmp.setPublicationDate(product.getPublicationDate());
            tmp.setPhotoGallery(product.getPhotoGallery());
            tmp.setCoordinates(product.getCoordinates());
            tmp.setMaxGuess(product.getMaxGuess());
            tmp.setRooms(product.getRooms());
            tmp.setServices(product.getServices());
            tmp.setMinNights(product.getMinNights());
            tmp.setMaxNights(product.getMaxNights());
            tmp.setScore(product.getScore());
            tmp.setReview(product.getReview());
            tmp.setPricePerNight(product.getPricePerNight());
            tmp.setProductAvailable(product.getProductAvailable());

            return productRepository.save(tmp);

        }else {
            return null;
        }

    }

    @Override
    public String delete(Integer id) {
        if(productRepository.findById(id).isPresent()){
            productRepository.deleteById(id);
            return "Product with id: " + id + " was deleted";
        }else{
            return "Product with id:" + id + " don't exist";
        }
    }

    public Product findProductByName(String name){
        return productRepository.findProductByProductName(name).orElse(null);
    }

    public List<String> getLocations(){
        return productRepository.getLocations();
    }

    public List<Product> getProductsByCategory(String category){
        return productRepository.getProductsByCategory(category);
    }

    public List<Product> getProductsByCity(String city){
        return productRepository.getProductsByCity(city);
    }

    public List<Product> getProductsByCategoryAndCity(String category, String city){
        return productRepository.getProductsByCategoryAndCity(category, city);
    }

    /* ==================== Constructor ======================= */

    @Autowired
    public ProductService(IProductRepository productRepository, CategoryService categoryService, CityService cityService, UserService userService) {
        this.productRepository = productRepository;
        this.categoryService = categoryService;
        this.cityService = cityService;
        this.userService = userService;
    }
}
