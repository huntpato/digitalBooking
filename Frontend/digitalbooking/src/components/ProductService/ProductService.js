import React from "react";
import Service2Icon from "../ReServices/Service2Icon"
import styles from "./ProductService.module.css";

export const ProductService = (props) => {
  const { services, container } = styles;
  const { productServices } = props;
console.log(props)
  return (
    <div className = {container}>
      {
        productServices.map((icon, index) =>{
          return ( 
            <div className={services} key={index}>
              <Service2Icon icon={icon} key={index} color= {{color: "var(--principal-color)"}} />
              <p>{icon[0].toUpperCase() + icon.substring(1)}</p>
            </div>
          )
        })
      }
    </div>
  );
};

ProductService.defaultProps ={

  productServices :["Wifi",
    "kitchen",
    " parking",
    "airconditioning"],
};