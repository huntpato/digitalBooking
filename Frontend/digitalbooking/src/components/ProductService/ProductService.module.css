.container{
    display:flex;
    flex-direction: row;
    /*background-color: tomato;*/
    justify-content: flex-start;
    flex-wrap: wrap;
    margin-bottom: 20px;
    width: 100%;
}

.services{
    display:flex;
    flex-direction: row;
    /*background-color: tomato;*/
    justify-content: flex-start;
    flex-wrap: wrap;
    margin-bottom: 40px;
    width: 25%;
}

.services p{
    margin-left: 15px;
}

@media (max-width: 768px){

    .container{
        display:flex;
        flex-direction: row;
        flex-wrap: wrap;
    }

    .services{
        display:flex;
        flex-direction: row;
        width: 50%;
    }

}

/*@media (max-width: 480px){

    .services{
        display:flex;
        flex-direction: row;
        background-color: tomato;
        width: 50%;
    }

}*/