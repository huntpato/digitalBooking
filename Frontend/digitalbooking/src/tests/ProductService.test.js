import React from 'react';
import { shallow } from 'enzyme';
import { ProductService } from '../components/ProductService/ProductService'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';


describe('Pruebas en componente <ProductService />', ()=> {
 
    const wrapper= shallow(<ProductService services={[
        "Wifi",
        "kitchen",
        " parking",
        "airconditioning"
    ]}/>);

    test('Comprobar que <ProductService /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })


    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div').at(0);
        expect(div.hasClass('container')).toBe(true);
    })
    

})
