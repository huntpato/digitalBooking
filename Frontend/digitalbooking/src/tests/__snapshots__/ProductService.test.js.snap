// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <ProductService /> Comprobar que <ProductService /> renderiza correctamente. 1`] = `
<div
  className="container"
>
  <div
    className="services"
    key="0"
  >
    <Service2Icon
      color={
        Object {
          "color": "var(--principal-color)",
        }
      }
      icon="Wifi"
      key="0"
    />
    <p>
      Wifi
    </p>
  </div>
  <div
    className="services"
    key="1"
  >
    <Service2Icon
      color={
        Object {
          "color": "var(--principal-color)",
        }
      }
      icon="kitchen"
      key="1"
    />
    <p>
      Kitchen
    </p>
  </div>
  <div
    className="services"
    key="2"
  >
    <Service2Icon
      color={
        Object {
          "color": "var(--principal-color)",
        }
      }
      icon=" parking"
      key="2"
    />
    <p>
       parking
    </p>
  </div>
  <div
    className="services"
    key="3"
  >
    <Service2Icon
      color={
        Object {
          "color": "var(--principal-color)",
        }
      }
      icon="airconditioning"
      key="3"
    />
    <p>
      Airconditioning
    </p>
  </div>
</div>
`;
